File: 진우/21weeks/13305. 주유소/주유소.cc
#include <iostream>
using namespace std;

long long dist[100001];
long long cost[100001];

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int N;
    cin >> N;

    for (int i = 0; i < N - 1; i++) cin >> dist[i];
    for (int i = 0; i < N; i++) cin >> cost[i];

    long long mn = cost[0];   // 지금까지의 최소 기름값
    long long total = 0;      // 총 비용

    for (int i = 0; i < N - 1; i++) {
        if (cost[i] < mn) mn = cost[i];
        total += mn * dist[i];
    }

    cout << total << "\n";
    return 0;
}

File: 진우/21weeks/13305. 주유소/주유소.py
import sys
input = sys.stdin.readline

N = int(input())

# 도시 사이 거리 (N-1개)
dist = list(map(int, input().split()))

# 각 도시의 기름 가격 (N개)
cost = list(map(int, input().split()))

mn = cost[0]   # 지금까지의 최소 기름값
total = 0      # 총 비용

# 마지막 도시는 이동이 없으므로 N-1까지만
for i in range(N - 1):
    if cost[i] < mn:
        mn = cost[i]
    total += mn * dist[i]

print(total)
